src/service/db/dashboards/mod.rs

```diff
@@ -13,111 +13,4 @@
 // You should have received a copy of the GNU Affero General Public License
 // along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
-use actix_web::web;
-use config::{
-    meta::dashboards::{v1, v2, v3, v4, v5, Dashboard, DashboardVersion, ListDashboardsParams},
-    utils::json,
-};
-use infra::table::dashboards;
-
 pub mod reports;
-
-#[tracing::instrument]
-pub(crate) async fn get(
-    org_id: &str,
-    dashboard_id: &str,
-    folder_id: &str,
-) -> Result<Option<Dashboard>, anyhow::Error> {
-    let dash = dashboards::get(org_id, folder_id, dashboard_id).await?;
-    Ok(dash)
-}
-
-#[tracing::instrument(skip(body))]
-pub(crate) async fn put(
-    org_id: &str,
-    dashboard_id: &str,
-    folder_id: &str,
-    body: web::Bytes,
-    hash: Option<&str>,
-) -> Result<Dashboard, anyhow::Error> {
-    if let Some(existing_dash) = get(org_id, dashboard_id, folder_id).await? {
-        let existing_dash_hash = existing_dash.hash;
-
-        let Some(Ok(hash_val)) = hash.map(|hash_str| hash_str.parse::<u64>()) else {
-            return Err(anyhow::anyhow!(
-                "Request to update existing dashboard with missing or invalid hash value. BUG"
-            ));
-        };
-        if hash_val.to_string() != existing_dash_hash {
-            return Err(anyhow::anyhow!(
-                "Conflict: Failed to save due to concurrent changes. Please refresh the page after backing up your work to avoid losing changes."
-            ));
-        }
-    };
-
-    let d_version: DashboardVersion = json::from_slice(&body)?;
-    let mut dash = parse_dashboard(body, d_version.version)?;
-
-    let title = dash
-        .title()
-        .map(|t| t.trim().to_string())
-        .and_then(|t| if t.is_empty() { None } else { Some(t) })
-        .ok_or_else(|| anyhow::anyhow!("Dashboard should have title"))?;
-    dash.set_title(title);
-
-    dash.set_dashboard_id(dashboard_id.to_owned());
-    let dash = dashboards::put(org_id, folder_id, dash).await?;
-    Ok(dash)
-}
-
-#[tracing::instrument]
-pub(crate) async fn list(params: ListDashboardsParams) -> Result<Vec<Dashboard>, anyhow::Error> {
-    let ds = dashboards::list(params).await?;
-    Ok(ds)
-}
-
-#[tracing::instrument]
-pub(crate) async fn delete(
-    org_id: &str,
-    dashboard_id: &str,
-    folder_id: &str,
-) -> Result<(), anyhow::Error> {
-    dashboards::delete(org_id, folder_id, dashboard_id).await?;
-    Ok(())
-}
-
-#[tracing::instrument]
-pub async fn reset() -> Result<(), anyhow::Error> {
-    dashboards::delete_all().await?;
-    Ok(())
-}
-
-/// Parses the bytes into a dashboard with the given version.
-pub(crate) fn parse_dashboard(
-    bytes: web::Bytes,
-    version: i32,
-) -> Result<Dashboard, serde_json::Error> {
```

Comment:
Moved into "HTTP Layer" since this deals specifically with parsing the REST HTTP request payload into a "Service Layer" domain model.